tests/trigger/base_test.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+
+
+class BaseTriggerTest(object):
+    @property
+    def trigger_func(self):
+        raise NotImplementedError(
+            "This method should be implemented by each concrete subclass"
+        )
+
+    @property
+    def good_input(self):
+        raise NotImplementedError(
+            "This method should be implemented by each concrete subclass"
+        )
+
+    @property
+    def exception_input(self):
+        raise NotImplementedError(
+            "This method should be implemented by each concrete subclass"
+        )
+
+    @property
+    def exception_raised(self):
+        return TypeError
```

Comment:
this is just nitpicky but I'm wondering why you chose `TypeError` here

Reply:
because as it turns out all of our existing triggers except deserialization raise the same thing :)

Reply:
iiiiiinteresting!